File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cs
public class Solution
{
	public int[] ProductExceptSelf(int[] nums)
	{
		int[] output = new int[nums.Length]; 
		int currentIndex = 0;
		
		for(; currentIndex < nums.Length; currentIndex++)
		{
			int left = 0;
			
			int sum = 1;
			while (left < currentIndex)
			{
				sum *= nums[left];
				left++;
			}
			
			int right = currentIndex + 1;
			
			while (right < nums.Length)
			{
				sum *= nums[right];
				right++;
			}
			
			output[currentIndex] = sum;
		}
		return output;
	}
}

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        min_length = len(nums) + 1

        buffer = []

        for i in nums:
            # Grow the buffer till sum >= target and then shrink from left till no longer >= target
            buffer.append(i)

            buffer_size = sum(buffer) 
            while (buffer_size >= target and len(buffer) > 0):
                min_length = min(len(buffer), min_length)

                # Remove from the top
                buffer.pop(0) 

                buffer_size = sum(buffer)

        if min_length == len(nums) + 1:
            return 0
            
        return min_length

        